simulator/func_sim/rf/rf.h

```diff
@@ -48,15 +47,14 @@ class RF
             hi_lo <<= 32;
             hi_lo += static_cast<uint64>(entry_lo.value);
             hi_lo += static_cast<uint64>(val);
-            new_val = static_cast<T>(hi_lo);
         }
         if ( num.is_mips_hi_lo()) {
-            write( Register::mips_hi, static_cast<uint64>(new_val) >> 32);
-            write( Register::mips_lo, new_val);
+            write( Register::mips_hi, static_cast<uint128>(val) >> 64);
```

Comment:
Ehh, it should be dependent on instruction type. Single word instructions still do shift to 32

Reply:
But why? we think that, yes, hi/lo now 64 bit, but we can put 32-bit value inside.

Reply:
It's OK, but then you do wrong shift anyway.

What MIPS32 expects is:
0x10000 * 0x10000 = 0x1 . 0x0 (dot is concatenation here)

Your shift does:
0x10000 * 0x10000 = 0x1 0000 0000
lo = 0x1 0000 0000
hi = 0x0

Reply:
And with div how id does? Sorry for late conversation

Reply:
`divu`: 9 / 4 = 1 . 2 (remainder . quotient) -> 0x1'0000'0002 in v_dst
`ddivu`: 9 / 4 = 1 . 2 -> 0x1'0000'0000'0000'0002 in v_dst